tests/common.go

```diff
@@ -354,17 +363,32 @@ func (ks *KubernetesSuite) logf(format string, args ...any) {
 }
 
 type logMatcher interface {
-	Match(reader io.Reader) (bool, error)
+	Match(reader io.ReadSeeker) (bool, error)
 	fmt.Stringer
 }
 
 // waitUntilLog waits until ctx expires or logs of container in all pods matching podLabels satisfy all logMatchers.
 func (ks *KubernetesSuite) waitUntilLog(ctx context.Context, namespace string, podLabels map[string]string, container string, description string, logMatchers ...logMatcher) {
 	ls := labels.SelectorFromSet(podLabels).String()
-	checkLogs := func() error {
-		podList, err := ks.k8s.CoreV1().Pods(namespace).List(ctx, metaV1.ListOptions{LabelSelector: ls})
+	checkLogs := ks.checkLogsClosure(ctx, namespace, ls, container, logMatchers...)
+	ks.logf("Waiting until %q pods logs "+description+": %s", ls, logMatchers)
```

Comment:
nit: since you're already using `logf`, then we should do `%s` for `description`

Reply:
Hrm, indeed, these logs all existed prior to this PR, I was mainly moving them around.